### ISSUE TITLE
Reference to localhost in invitations

### ISSUE THREAD

**Issue by Speirs47** [flyve-mdm/glpi-plugin#791]
**Describe the bug**
Whenever I send an invitation to enroll my phone it tries to connect localhost instead of IP address specified in General options or API options

**To Reproduce**
1. Send an invitation.
2. Scan the invitations QR Code with FlyveMDM Agent (or open link in this app)
3. Wait for the Agent to create inventory
4. See error mentioning failed connection to localhost like the one on attached screenshot

**Expected behavior**
Well, I think it shouldn't try to connect to localhost but to the host specified in GLPI config.

**Screenshots**
![screenshot_20181114-103211](https://user-images.githubusercontent.com/40421338/48473431-bc307c80-e7f8-11e8-8395-6f435bf608b1.png)

**Desktop:**
 - OS: openSUSE Leap 15 (GLPI in docker)
 - Browser Firefox
 - Version 60.3.0 ESR

**Smartphone:**
 - Device: Sony Xperia XA2
 - OS: Android 8.0.0
 - Browser Firefox
 - Version 63.0.2

**Comment by DIOHz0r** [comment]
Hi,

The URL for invitation comes from your GLPI configuration. You have first check your "URL of the application" value, it's on Setup -> General options. 

That URL must be accessible from anywhere inside your network (or to the Internet if your server is on the cloud), so the URL access must be public and resolved for your mobile devices.

**Comment by Speirs47** [comment]
![obraz](https://user-images.githubusercontent.com/40421338/48553526-1ce4b580-e8dc-11e8-8d35-cf7cb91d5456.png)

Here's my "URL of the application". I can access GLPI via browser on my phone, even send the invitation but I can't enroll my phone because it says it can't connect to localhost:8080 as mention on screenshot I attached before.

**Comment by DIOHz0r** [comment]
Hi, sorry for the confusion but the URL comes from the API URL at Setup -> General  -> API

**Comment by Speirs47** [comment]
Same address here
![obraz](https://user-images.githubusercontent.com/40421338/48554880-e5780800-e8df-11e8-93e4-d754d992e915.png)

**Comment by DIOHz0r** [comment]
I can't reproduce this problem, I was able to go to the registration form with an invitation using the static IP and port of the server. If I change the URL of the API the invitation is sent with the correct data for access to the server. 

Are you using an old invitation link or do you create a new invitation after setting up the URL for your API?. Another issue could be that your device has an invalid registry on its "/system/etc/hosts" file setting the IP 192.168.10.176 as localhost

**Comment by Speirs47** [comment]
I create a new invitation everytime I try to test it (although it seems that the link is always the same, but I didn't verify whole string). Hosts file is set OK.

Maybe it's something wrong with docker image I used (I didn't use the one you got there on github). I will try to install GLPI, fusioninventory and flyvemdm without docker.

**Comment by Speirs47** [comment]
Ok, I get it, it's a bug but with Android app. 

When I tried enrolling for the first time I did forget to change **URL of the application** and **API URL** so the error mentioned here was correct. After I changed it, it still tried to connect to localhost because it wasn't trying to read the correct host from the QR code but used some cached information instead.

After I wipe Agent app data and tried to enroll again, everything worked just fine.
